Bio/PDB/Mutate.py

```diff
@@ -0,0 +1,578 @@
+                *[sample_residue[x] for x in CHI_ANGLES[angle][mutate_to]["ref_plane"]]
+            )
+            rotation_angle = dihedral_start - np.deg2rad(rotamer[angle])
+            axis = CHI_ANGLES[angle][mutate_to]["axis"]
+            # print(angle)
+            for atom in RESIDUE_ORDER[mutate_to][
+                RESIDUE_ORDER[mutate_to].index(axis[1]) + 1 :
+            ]:
+                sample_residue[atom] = (
+                    np.dot(
+                        rotation_matrix(
+                            sample_residue[axis[0]] - sample_residue[axis[1]],
+                            rotation_angle,
+                        ),
+                        sample_residue[atom] - sample_residue[axis[1]],
+                    )
+                    + sample_residue[axis[1]]
+                )
+
+        for rotamer_atom, rotamer_vector in sample_residue.items():
+            if vdw_energy > lowest_energy:  # Skip pointless rotamers
+                break
+            atoms = unfold_entities(pdb_object[0], "A")
+            ns = NeighborSearch(atoms)
+            close_atoms = ns.search(rotamer_vector, 5)  # 5 Angstrom radii
+            for close_atom in close_atoms:
+                close_residue = close_atom.get_parent()
+                chain_if_close_atom = close_residue.get_parent()
+                if (
+                    close_atom.get_parent().get_id()[1] == res_num
+                    and chain_if_close_atom.get_id() == chain
+                ):
+                    # print("skipping_own_atom")
+                    continue
+
+                if skip_own_chain and chain_if_close_atom.get_id() == chain:
+                    # print("Skipping own chain")
+                    continue
+                if abs(close_residue.get_id()[1] - res_num) == 1 and is_backbone(
+                    close_atom
+                ):
+                    continue
+                dist = distance(close_atom.coord, rotamer_vector)
+                if dist > 6:
+                    continue
+                try:
+                    vdw_radi = (
+                        VW_RADII[close_atom.get_parent().get_resname()][
+                            close_atom.get_id()
+                        ]
+                        + VW_RADII[mutate_to][rotamer_atom]
+                    )
+                except KeyError:
+                    continue
+                attractive_force = (vdw_radi / dist) ** 6
+                vdw_energy += attractive_force**2 - attractive_force
+        if vdw_energy < lowest_energy:
+            lowest_energy = vdw_energy
+            best_rotamer = rotamer
+    return best_rotamer
+
+
+def mutate(
+    pdb_obj,
+    chain,
+    res_num,
+    mutate_to,
+    rotamer_lib=None,
+    mutation_type="best",
+    verbose="info",
+):
+    """
+    Mutates a given residue based on the best fitting rotamers
+    """
+    level = logging.getLevelName(verbose.upper())
+    logging.getLogger().setLevel(level)
```

Comment:
It feels a little bit dangerous to change the level of the root logger in this function (especially by default). Some unsuspecting person could import our library, use this function, and suddenly the logging they set up gets changed. Instead, if you want to use the logging module, you probably want to set up a file-specific logger like:
`logger = logging.getLogger(__name__)`, then you can mess with its logging level without affecting the logging of others

Reply:
See the logging module for more details: https://docs.python.org/3/library/logging.html

Reply:
Do you need the logger at all? We don't use this elsewhere in Biopython that I recall.